CONTRIBUTING.md

```diff
@@ -129,10 +129,33 @@ To start che-theia in `dev-mode` with yarn (not using the production che-theia g
 Running dev theia would be located in the che-dev container and `theia-dev-flow` endpoint:
 ![Che-Theia-dev-endpoint](https://raw.githubusercontent.com/eclipse/che-theia/assets/theia-dev-flow-endpoint.png)
 
-
-
 ### Just want to build the plugin and run with the existing Theia image
 If you do not have any changes on Theia or Che-theia extension, you could just build the plugins with `build ... containers-plugin` or `build ... factory-plugin`
 and run these plugins with the existing che-theia app:
 `run ... HOSTED che-theia + container-plugin` or `run ... HOSTED che-theia + factory-plugin`
 
+## Contributing to other parts of Che Theia
+
+### How to develop Che Theia remote plugin mechanism
+
+_Please note, this section provides a flow how to develop remote plugin mechanism in Che Theia, but not a remote plugin._
+
+First, create a workspace from prepared [devfile](https://github.com/eclipse/che-theia/blob/master/extensions/eclipse-che-theia-plugin-remote/devfile.yaml), which could be found in the `eclipse-che-theia-plugin-remote` extension folder.
+When workspace is ready:
+ - Init Che Theia. This could be done with `che:theia init` command in `/projects/theia` folder or run the init command.
+   Che Theia sources will be awailable at `/projects/theia/che/che-theia`.
+ - Now one may make changes in Che Theia remote plugin mechanism in both (Che Theia and Remote plugin) sides.
+ - Build Che Theia from `theia-dev` container by executing `yarn` in `/projects/theia` folder or by running corresponding command.
+ - Put your remote plugin(s) into `/projects/remote-plugins/` directory.
+   Note that the plugin(s) shouldn't have any external dependencies.
+   For example, [this sample plugin](https://github.com/eclipse/che-theia-samples/tree/master/samples/hello-world-backend-plugin) might be used.
```

Comment:
to me it would be better to add that sample in the devfile (as a project to clone) so there is no doubt about what to do to make it work.

Reply:
I think it is not needed. A developer can generate a plugin and add simple functionality to test his/her changes. No need in cloning all the samples to have hello world backend plugin. It is faster to generate one. At the end, there is a link to sample...

Reply:
`local-dir:///projects/remote-plugins/` is not clear if the source code should be there or the binaries or folders of source code. This is why having the project cloned makes things more clear and it is easy to adapt the devfile afterwards.

Reply:
But it is the same... cloned repository won't clear the situation. To me it will be even more unclear if we clone samples.
What should be changed - it's the description. I agree we can add the remark.

Reply:
so this is about developing che-theia remote plugin mechanism. We don't mind if this is a sample or not it would just be for testing. User could change it if he would like to.

Reply:
@sunix I am still not keen on the idea to clone all the samples, because they have many frontend plugins which cannot be used here. Also it is faster to generate a new plugin than clone all the samples and find needed to change/compile.